avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -255,15 +258,18 @@ private Schema getSpecificReaderSchema(Schema writerSchema) {
   }
 
   private Schema getReflectionReaderSchema(Schema writerSchema) {
-    // shall we use ReflectData.AllowNull.get() instead?
-    Class<?> readerClass = ReflectData.get().getClass(writerSchema);
+    Class<?> readerClass = avroReflectionAllowNull
```

Comment:
Let's declare `ReflectData reflectData = avroReflectionAllowNull ? ReflectData.AllowNull.get() : ReflectData.get()` and then use `reflectData` throughout this method.